### ISSUE TITLE
libsndfile for .NET

### ISSUE THREAD

**Issue by aybe** [erikd/libsndfile#347]
I'm about to finish a wrapper for .NET, I've implemented everything but opening a file descriptor since it crashes no matter what I try.

Do you happen by chance to have some test files I can throw at it ?
(I am looking for files containing these special chunks : RF64, Cart and so on)

Thanks !

**Comment by evpobr** [comment]
Do not implement. It is [documented](http://www.mega-nerd.com/libsndfile/api.html#open_fd) that `sf_open_fd` works only under POSIX systems.

Passing CRT objects (`FILE*`, file descriptor) under Windows is [bad idea](https://msdn.microsoft.com/en-us/library/ms235460.aspx).

I've got strange tests fails with `libsndfile-1.dll` all the time, when static library works fine.

**Comment by aybe** [comment]
Yes I've learned the hard way, dotnet.exe process (.NET core command-line app) simply crashes without any further details, I've just removed that functionality entirely.

The `sf_open_virtual` works, is enough and will actually [come in handy under UWP](https://trac.ffmpeg.org/wiki/CompilationGuide/WinRT#WindowsStoreCertificationFileIOandOtherDetails) where you cannot open 'whatever` file you like with legacy IO.

**Edit**

Actually I don't know if virtual IO would ever be needed for UWP since [I committed this patch to vcpkg](https://github.com/aybe/vcpkg/commit/72fdd294b05f7b29cc711757a819df2948c72d80), I just got the thing to compile but still need to test it ... at least for the vcpkg build I'm using.

I'll soon push my repo so you guys can take a look and let me know what you think (still testing :).

**Comment by aybe** [comment]
That was a bit longer than expected but here it is !
https://github.com/aybe/libsndfile.NET

Now I will start using it and fix potential bugs as I find them, basic functionality works but things like reading special tags needs testing (specifically the marshalling part), I need to get hold of some files with these chunks.

Cheers :)
